<?xml version="1.0"?>
<launch>
    <arg name="description_for_moveit" default="/my_gen3/robot_description"/>
    <arg name="launch_apriltag" default="false"/>
    <arg name="launch_gripper" default="true"/>
    <arg name="launch_ft_sensor" default="true"/>

    <!-- Remap description for moveit commander -->
    <param name="/moveit_planner/description_name" value="$(arg description_for_moveit)"/>
    <param name="/moveit_planner/namespace" value="my_gen3"/>
    <param name="/moveit_planner/arm_group_name" value="arm"/>

    
    <!--    Force sensor and filtering/compensation relay node-->
    <group if="$(arg launch_ft_sensor)">
      <include file="$(find sensor_tools)/launch/ft_sensor_standalone.launch"/>
    <!--   <node name="reflex_controller_spawner" pkg="controller_manager" type="spawner" respawn="false"
               output="screen" args="reflex_controller"/> -->
    </group>
    
    <!--    Gripper control-->
    <group if="$(arg launch_gripper)">
        <node name="robotiq_2f_gripper_control" pkg="robotiq_2f_gripper_control" type="Robotiq2FGripperRtuNode.py" args="/dev/r2Fg" output="screen"/>
        <node name="robotiq_2f_gripper_commander" pkg="kinova_valve_opening" type="gripper_commander.py" output="screen"/>
    </group>

    
    <!--   Apriltag detection -->
    <group if="$(arg launch_apriltag)">
        <include file="$(find kinova_valve_opening)/apriltag/apriltag_detection.launch"/>
    </group>
    <group unless="$(arg launch_apriltag)">
<!--         <node name="valve_broadcaster" type="static_transform_publisher" pkg="tf2_ros"
              args="0.0 0.46 0.125 0.707 0 0 0.707 arm_base_link object_detection_mesh"/> -->
        <node name="valve_broadcaster" type="static_transform_publisher" pkg="tf2_ros"
              args="0.0 0.46 0.125  -0.6830127 0.1830127 0.1830127 0.6830127 arm_base_link object_detection_mesh"/>
              
    </group>
    

</launch>